gpt_bpe_test.go

```diff
@@ -1015,6 +1052,93 @@ func TestModelDownloadPythiaSharded(t *testing.T) {
 
 }
 
+func TestModelDownloadLlama(t *testing.T) {
+	// Pythia uses a slightly different file structure, where
+	// the vocab.json and merges.txt files are stored in the
+	// tokenizer.json file. We want to check if we are able to
+	// download the model and extract the vocab.json and merges.txt
+	modelId := "georgesung/llama2_7b_chat_uncensored"
+	destPath := "./TestModelDownloadLlama"
+	destPathPTR := &destPath
+	os.RemoveAll(destPath)
+
+	var rsrcType resources.ResourceType
+	rsrcType = resources.RESOURCETYPE_TRANSFORMERS
+	hfApiToken := os.Getenv("HF_API_TOKEN")
+	os.MkdirAll(destPath, 0755)
+	_, rsrcErr := resources.ResolveResources(modelId, destPathPTR,
+		resources.RESOURCE_MODEL, rsrcType, hfApiToken)
+	if rsrcErr != nil {
+		//os.RemoveAll(destPath)
+		t.Errorf("Error downloading model resources: %s", rsrcErr)
+	}
+
+	// Check that the model files are there
+	// We want to check for the presence of the following files:
+	// config.json, pytorch_model.bin,
+	// tokenizer.json, vocab.json
+
+	// Check for config.json
+	configPath := destPath + "/config.json"
+	if _, err := os.Stat(configPath); err == nil {
+		fmt.Println("config.json exists")
+
+	} else if errors.Is(err, os.ErrNotExist) {
+		//os.RemoveAll(destPath)
+		t.Errorf("config.json does not exist")
+
+	} else {
+		//os.RemoveAll(destPath)
+		t.Errorf("Error checking for config.json")
+	}
```

Comment:
It would be useful for debugging if `err` itself was passed. There's probably not a lot of encounterable errors with something like this, but it's still always better to pass in `err` rather than an abstract message like this